### PR TITLE
savedata_factory: Automatically create certain savedata

### PR DESCRIPTION
**Pull request by DarkLordZach**
After further hardware investigation, it appears that some games, perhaps those more lazily coded, will not call EnsureSaveData, meaning that they expect the normal (current) save to be automatically made. Additionally, some games do not create a cache or temporary save before use.
In these 3 specific instances, the save is created automatically for the game if it doesn't exist.

### CODE REVIEW

**Comment by DarkLordZach** [comment]
Closes #2999 & #2963

File: src/core/file_sys/savedata_factory.cpp [hunk 1]
```diff
@@ -52,6 +53,14 @@ void PrintSaveDataDescriptorWarnings(SaveDataDescriptor meta) {
                     meta.user_id[1], meta.user_id[0]);
     }
 }
+
+bool ShouldSaveDataBeAutomaticallyCreated(SaveDataSpaceId space, const SaveDataDescriptor& desc) {
+    return desc.type == SaveDataType::CacheStorage        ///< Cache Saves
+           || desc.type == SaveDataType::TemporaryStorage ///< Temp Saves
```
Comment (lioncash) [inline src/core/file_sys/savedata_factory.cpp]: Our convention is to place the operator on the other side of the expression. The first two comments are also a little bit superfluous (the enum types themselves adequately describe the type of save)

**Comment by Morph1984** [comment]
Looks like this solution does not address a failed stat on the temp directory since this particular game (SNES Online) calls stat on the temp directory before mounting the directory with `MountTemporaryStorage`, assuming that this directory existed prior to calling stat or opening it.

```
[  22.293478] Service.FS <Debug> core\hle\service\filesystem\fsp_srv.cpp:OutputAccessLogToSdCard:1244: called, log='FS_ACCESS: { start:       187, end:       187, result: 0x00000000, handle: 0x0000000000000000, priority: Normal, function: "MountRom", name: "rom" }
[  22.297002] Common.Filesystem <Debug> common\file_util.cpp:IsDirectory:128: stat failed on C:/Users/User/AppData/Roaming/yuzu/nand/temp/: The system cannot find the file specified.
[  22.297170] Common.Filesystem <Debug> common\file_util.cpp:IsDirectory:128: stat failed on C:/Users/User/AppData/Roaming/yuzu/nand/temp/0000000000000000/: The system cannot find the file specified.
[  22.297276] Common.Filesystem <Debug> common\file_util.cpp:IsDirectory:128: stat failed on C:/Users/User/AppData/Roaming/yuzu/nand/temp/0000000000000000/00000000000000000000000000000000/: The system cannot find the file specified.
[  22.298966] Service.FS <Debug> core\hle\service\filesystem\fsp_srv.cpp:OutputAccessLogToSdCard:1244: called, log='FS_ACCESS: { start:       187, end:       187, result: 0x00000000, handle: 0x0000000000000000, priority: Normal, function: "MountTemporaryStorage", name: "temp" }
```

**Comment by DarkLordZach** [comment]
@Morph1984 Can you clarify whether or not this causes the game to softlock?

**Comment by Morph1984** [comment]
It does not softlock

**Comment by DarkLordZach** [comment]
Then the failed stat is from yuzu checking for the directories existence before attempting to create it.

Thus, no issue

**Pull request merged**